### PR TITLE
fix: correct some ts errors

### PR DESCRIPTION
**Pull request by chambo-e**
## Type

- Enhancement

### CODE REVIEW

**Review by philibea — APPROVED** [review]

**Review by matthprost — APPROVED** [review]

File: src/components/RichSelect/__stories__/index.stories.tsx [hunk 1]
```diff
@@ -52,9 +52,13 @@ Uncontrolled.decorators = [
...
 Controlled.decorators = [
   () => (
-    <ControlValue value="">
+    <ControlValue<string> value="">
       {({ value, onChange }) => (
-        <RichSelect name="controlled" value={value} onChange={onChange}>
+        <RichSelect
+          name="controlled"
+          value={value}
+          onChange={newValue => newValue && onChange(newValue)}
```
Comment (QuiiBz) [inline src/components/RichSelect/__stories__/index.stories.tsx:60]:
```suggestion
          onChange={newValue => newValue && onChange(newValue) as unknown as string}
```

Comment (chambo-e) [reply src/components/RichSelect/__stories__/index.stories.tsx:60]: Meh, force casting should not be the solution right ?

Comment (QuiiBz) [reply src/components/RichSelect/__stories__/index.stories.tsx:60]: This is what you've done here: https://github.com/scaleway/scaleway-ui/pull/1378/files#diff-401752b10b3fb35cfa63400a966be9aef2b40e276df59abfa2007d1065136623R132

Btw I made a mistake, I wanted to type cast `newValue`, not `onChange`

File: src/components/List/__stories__/index.stories.tsx [hunk 2]
```diff
@@ -525,12 +525,19 @@ PaginationLoadingPageCount.decorators = [
...
-        const [options] = useState(
+        const [options] = useState<
+          { label: keyof typeof animations; value: keyof typeof animations }[]
+        >(
           Object.keys(animations).map(animation => ({
-            label: animation,
-            value: animation,
+            label: animation as keyof typeof animations,
+            value: animation as keyof typeof animations,
           })),
```
Comment (Exayy) [inline src/components/List/__stories__/index.stories.tsx:541]: What about (one typing only I don't know if it clearer or not) :
```
(Object.keys(animations) as (keyof typeof animations)[]).map(
            animation => ({
              label: animation,
              value: animation,
            }),
          ),
```

**Review by Exayy — APPROVED** [review]

File: src/components/Menu/Item.tsx [hunk 3]
```diff
@@ -31,11 +32,45 @@ const variantStyle = {
...
+  borderless?: boolean
+  variant?: VariantItem
+}
+
+const StyledButton = styled(Button)<{ borderless: boolean }>`
+  display: inline-block;
+  font-size: 14px;
+  line-height: 22px;
+  font-weight: inherit;
+  padding: 4px 8px;
```
Comment (QuiiBz) [inline src/components/Menu/Item.tsx:47]: Theme can be used for the padding

**Pull request closed without merging**